File: packages/shared-types/order.ts
export type OrderStatus = 'pending' | 'paid' | 'processing' | 'partially_shipped' | 'completed' | 'cancelled';
export type SubOrderStatus = 'placed' | 'confirmed' | 'in_crafting' | 'shipped' | 'delivered' | 'cancelled' | 'refunded';

export interface OrderItemPriceBreakdown {
  base_price: number;
  artisan_share: number;
  platform_fee: number;
  delivery_fee: number;
}

export interface OrderItem {
  id: string;
  suborder_id: string;
  product_variant_id: string;
  product_title: string;
  variant_name: string;
  quantity: number;
  unit_price: number;
  price_breakdown: OrderItemPriceBreakdown;
}

export interface SubOrder {
  id: string;
  order_id: string;
  artisan_id: string;
  artisan_name: string;
  subtotal: number;
  status: SubOrderStatus;
  items: OrderItem[];
  tracking_number?: string;
  carrier?: string;
  created_at: string;
}

export interface Order {
  id: string;
  user_id: string;
  total_amount: number;
  status: OrderStatus;
  suborders: SubOrder[];
  razorpay_order_id?: string;
  created_at: string;
}

File: backend/app/schemas/order.py
from typing import List, Optional
from datetime import datetime
from pydantic import BaseModel

class OrderItemBreakdownSchema(BaseModel):
    base_price: float
    artisan_share: float
    platform_fee: float
    delivery_fee: float

class OrderItemSchema(BaseModel):
    id: str
    suborder_id: str
    product_variant_id: str
    product_title: str
    variant_name: str
    quantity: int
    unit_price: float
    total_price: float
    price_breakdown: Optional[OrderItemBreakdownSchema] = None

    class Config:
        from_attributes = True

class SubOrderSchema(BaseModel):
    id: str
    suborder_number: str
    artisan_id: str
    artisan_name: str
    subtotal: float
    artisan_earnings: float
    platform_commission: float
    delivery_fee: float
    status: str
    items: List[OrderItemSchema] = []
    created_at: datetime

    class Config:
        from_attributes = True

class OrderCheckoutRequest(BaseModel):
    address_id: Optional[str] = None

class OrderSchema(BaseModel):
    id: str
    order_number: str
    user_id: str
    total_amount: float
    currency: str
    status: str
    razorpay_order_id: Optional[str] = None
    suborders: List[SubOrderSchema] = []
    created_at: datetime

    class Config:
        from_attributes = True
